actix-http/src/error.rs

```diff
@@ -55,6 +55,8 @@ impl Error {
         Self::new(Kind::Io)
     }
 
+    // used in encoder behind feature flag so ignore unused warning
+    #[allow(unused)]
     pub(crate) fn new_encoder() -> Self {
```

Comment:
Is there any chance this will be used in the future? If so that's fine, but if not, then I'd prefer `#[cfg(..)]` or `#[cfg_attr()]` to it (but yeah, the current may be a simple way).

Reply:
To be clear, it is used in current code but only by modules behind "__compress" feature.

I don't really like the idea of changing this error type based on the feature flags, potentially makes other things more difficult in downstream crates. It's free to keep all error variants in place even if their feature is disabled given how it works.